dxp/how_clay_arrives_to_dxp.md

```diff
@@ -0,0 +1,223 @@
+| Main CSS of the Liferay administration panels             |          |        |         | ✅    |
+
+From a more technical point of view, setting the base through the generator will create a setting named `liferayTheme` in the `package.json` file of the theme folder.
+
+```json
+{
+	"name": "my-liferay-theme",
+	"version": "1.0.0",
+	"main": "package.json",
+	"keywords": ["liferay-theme"],
+	"liferayTheme": {
+		"baseTheme": "styled",
+		"templateLanguage": "ftl",
+		"version": "7.0"
+	},
+	"devDependencies": {
+		"gulp": "^3.8.10",
+		"liferay-theme-deps-7.1": "*",
+		"liferay-theme-tasks": "*"
+	}
+}
+```
+
+[Source](https://github.com/liferay/liferay-js-themes-toolkit/tree/master/packages/liferay-theme-tasks#liferaytheme)
+
+## How to import Clay into a custom-theme?
+
+We always recommend to use [liferay-js-themes-toolkit](https://github.com/liferay/liferay-js-themes-toolkit) to create or update a theme, but we understand that sometimes you could have the need of adding Clay to an old heavy-customized theme that can't be automatically updated. In those cases we will have to import Clay as a depencency.
+
+The first step will be to localize the project's folder and install the right dependencies. You can either use the classic Node system or the modern Yarn one.
+
+Node
+`npm install @clayui/css`
+
+Yarn
+`yarn add @clayui/css`
+
+It's important to know that these commands will only install the CSS library, if you are building a theme that can make a good use of Clay components created in React, we recommend to use:
+
+Node
+`npm install @clayui/*`
+
+Yarn
+`yarn add @clayui/*`
+
+You can also address specific components with commands like `npm install @clayui/buttons`, `yarn add @clayui/cards`, `yarn add @clayui/tables@^version`, etc. [Read more at Clay's website](https://clayui.com/docs/get-started/how-to-use-clay.html)
+
+## How is Clay imported into a module?
+
+Unlike the previous section we won't need to install Clay as a dependency in a module inside Liferay Portal. Thanks to our rich environment, we already have it included and we just need to enable it.
+
+To do so, we need to define the new import into our dedicated buildCSS process, we can do that copy the following code into the `build.gradle` file of our module.
+
+```js
+buildCSS {
+	imports = [
+		new File(npmInstall.nodeModulesDir, "@clayui/css/src/scss")
+	]
+}
+```
+
+[Real example of buildCSS](https://github.com/liferay/liferay-portal/blob/master/modules/apps/adaptive-media/adaptive-media-web/build.gradle)
+
+## How to import Clay into a custom-module?
+
+Before unreavel the final piece of this first part, we must understand what's a module and what's the difference with a theme.
+
+In Liferay we refer to a module when we talk about an indipendent bundle that can be install in DXP. It usually contains configuration, markup and style. We often call them Widgets, Portlets or Themes.
```

Comment:
indipendent => independent
install => installed

Reply:
I think we should settle these naming concepts and maybe provide a limited thesaurus/taxonomy. I will send a PR to this same repo to try to agree on them.

I say this because we are using _module_ for almost everything, _package_ for several different things and _bundle_ for different things depending on the conext (just to name a few).

Reply:
> I will send a PR to this same repo to try to agree on them.

Looking forward to it...

![evil-waiting](https://user-images.githubusercontent.com/7074/94787873-280f3480-03d3-11eb-88ac-f6bb33d51e16.gif)

> maybe provide a limited thesaurus/taxonomy

Let's call it a _glossary_.

Reply:
Here it is -> https://github.com/liferay/liferay-frontend-guidelines/pull/224